shell/server/pre-meteor.js

```diff
@@ -210,7 +210,24 @@ function checkMagic(buf, magic) {
   return true;
 }
 
-function serveStaticAsset(req, res) {
+function serveSelfTest(req, res) {
+  inMeteor(() => {
+    if (req.method === 'GET' &&
+        req.url === '/') {
+      return serveStaticAsset(req, res, {
```

Comment:
Oh I see you were using a separate subdomain all along, so my comment was silly. But reusing the code to handle OPTIONS still seems unnecessary since GETs don't require preflights...